torch/testing/_internal/opinfo/core.py

```diff
@@ -946,6 +958,20 @@ def __post_init__(self):
                 else self.dtypes
             )
         )
+
+        self.backward_dtypesIfXPU = (
+            set(self.backward_dtypesIfXPU)
+            if self.backward_dtypesIfXPU is not None
+            else (
+                self.backward_dtypes
+                if self.backward_dtypes is not None
+                else self.dtypesIfXPU
+                if self.dtypesIfXPU is not None
+                #else self.dtypes
```

Comment:
Why do we need to skip to return `self.dtypes`?

Reply:
by default to align with cuda, so if dtypesIfXPU is None will return dtypesIfCuda. This is also to align with inductor test. https://github.com/pytorch/pytorch/blob/b5c2f6b2568bb2fabbdbb6aad63525b00e03a2ab/torch/testing/_internal/opinfo/core.py#L972 .

Reply:
If the `self.dtypes` is invalid now, should we remove the line of code?

Reply:
Updated, please review